core/src/main/java/org/jruby/RubyHash.java

```diff
@@ -307,13 +306,13 @@ public RubyHash(Ruby runtime, Map valueMap, IRubyObject defaultValue) {
     }
 
     private final void allocFirst() {
-        entries = new RubyHashEntry[MRI_INITIAL_CAPACITY];
-        bins = new int[MRI_INITIAL_CAPACITY * 2];
+        entries = new IRubyObject[MRI_INITIAL_CAPACITY * 2];
+        bins = new int[MRI_INITIAL_CAPACITY * 4];
```

Comment:
This might not make sense. The entries array is now twice as big but is also twice as fast full (because we store key and value in separate rows now). So I think we could still go with MRI_INITIAL_CAPACITY * 2 for the bins array.